draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -510,39 +510,79 @@ the salt and sizes to fit this form.
+nominal group as the traditional component.  This common structure uses one of
+two "combiner" functions, which determines the security requirements for the
+post-quantum KEM in order for the hybrid KEM to be secure.
 
-GHP:
-: A generic framework that is suitable for use with any choice of PQ and
-  traditional KEMs, with minimal security assumptions on the constituent KEMs
+To instantiate this framework, a designer will have to make two choices:
 
-QSF:
-: An optimized generic framework for the case where the PQ component has strong
-  binding properties and the traditional component is a nominal group
+1. Whether traditional component is a KEM or a nominal group
+2. Whether the post-quantum component assumed to satisfy the C2PRI property
 
-These frameworks share a common overall structure, differing mainly in how they
-compute the final shared secret and the security requirements of their
-components.
+The former choice will dictate the overall structure of the hybrid KEM, since
+traditional KEMs and nominal groups have different operations.  The latter
+choice will determine which combiner function should be used.
 
-## GHP {#ghp}
+## Combiner Functions
 
-The GHP hybrid KEM depends on the following constituent
-components:
+A combiner function uses the KDF used in the hybrid KEM to combine the shared
+secrets output by the constituent algorithms with metadata.  This operation is
+important to the IND-CCA security of the hybrid KEM, as well as its binding
+properties.
+
+The two combiner functions defined in this document are as follows:
+
+```
+def Explicit(ss_PQ, ss_T, ct_PQ, ct_T, ek_PQ, ek_T, label):
+    KDF(concat(ss_PQ, ss_T, ct_PQ, ct_T, ek_PQ, ek_T, label))
+
+def PQImplicit(ss_PQ, ss_T, ct_PQ, ct_T, ek_PQ, ek_T, label):
+    KDF(concat(ss_PQ, ss_T, ct_T, ek_T, label))
+```
+
+Note that while the names of the inputs are suggestive of the shared secret,
+ciphertext, and encapsulation key outputs of a KEM, the inputs to this function
+in the hybrid KEM framework are not necessarily the output of a secure KEM.  In
+particular, when the framework is instantiated with a nominal group, the
+"ciphertext" component is an ephemeral group element, and the "encapsulation
+key" is the group element that functions as the recipient's public key.
+
+As discusssed in {{security-requirements}}, the choice of combiner in a given
+instantitation determines the assumptions under which the resulting hybrid KEM
+is secure.
+
+The `Explicit` combiner explicitly hashes shared secrets, ciphertexts, and
+encapsulation keys from both constituent.  This allows the resulting hybrid KEM
+to be secure as long as either component is secure, with no further assumptions.
+
+The `PQImplicit` combiner does not explicitly hash in the ciphertext or
+encapsulation key from the PQ constituent.  The resulting hybrid KEM will only
+be secure if, in addition to one constituent being secre, the PQ constituent
+also satisfies the C2PRI property.
```

Comment:
I do not think this simplification / uniform API helps us much and seems squishier and more unclear than explicit generic frameworks; we can explicitly write them all (4?) down and that's it